pkg/resources/cruisecontrol/configmap.go

```diff
@@ -28,46 +28,81 @@ import (
 	"github.com/banzaicloud/kafka-operator/pkg/util"
 	kafkautils "github.com/banzaicloud/kafka-operator/pkg/util/kafka"
 	zookeeperutils "github.com/banzaicloud/kafka-operator/pkg/util/zookeeper"
+	properties "github.com/banzaicloud/kafka-operator/properties/pkg"
 
 	corev1 "k8s.io/api/core/v1"
 )
 
 func (r *Reconciler) configMap(clientPass, capacityConfig string, log logr.Logger) runtime.Object {
+	ccConfig := properties.NewProperties()
+
+	// Add base Cruise Control configuration
+	conf, err := properties.NewFromString(r.KafkaCluster.Spec.CruiseControlConfig.Config)
+	if err != nil {
+		log.Error(err, "parsing Cruise Control configuration failed", "config", r.KafkaCluster.Spec.CruiseControlConfig.Config)
+	}
+	ccConfig.Merge(conf)
+
+	// Set bootstrap servers
+	bootstrapServers := fmt.Sprintf("%s:%d",
+		generateBootstrapServer(r.KafkaCluster.Spec.HeadlessServiceEnabled, r.KafkaCluster.Name),
```

Comment:
[this is not related to the patch]

Here we manually generate the server address from the available information. 

@baluchicken I think we should remove these places from the code base where we manually assemble the server address: we should use the listener status field wherever possible - or at least from the components that depend on Kafka. By the time the reconcile gets there, the listener statuses should have been populated and we can use that field in CC.

Reply:
I agree we should do this in a separate PR:)